frontend/storm_wiki/util/storm_runner.py

```diff
@@ -0,0 +1,472 @@
+            "description": snippet,
+        }
+    return citations
+
+
+def process_search_results(runner, current_working_dir: str, topic: str):
+    topic_dir = os.path.join(current_working_dir, topic.replace(" ", "_"))
+    raw_search_results_path = os.path.join(topic_dir, "raw_search_results.json")
+    markdown_path = os.path.join(topic_dir, f"{topic.replace(' ', '_')}.md")
+
+    if os.path.exists(raw_search_results_path):
+        try:
+            with open(raw_search_results_path, "r") as f:
+                raw_search_results = json.load(f)
+
+            citations = process_raw_search_results(raw_search_results)
+            add_citations_to_markdown(markdown_path, citations)
+            logger.info(f"Citations added to {markdown_path}")
+        except json.JSONDecodeError:
+            logger.error(f"Error decoding JSON from {raw_search_results_path}")
+        except Exception as e:
+            logger.error(f"Error processing search results: {str(e)}", exc_info=True)
+    else:
+        logger.warning(f"Raw search results file not found: {raw_search_results_path}")
+
+
+def add_citations_to_markdown(markdown_path: str, citations: Dict[int, Dict[str, str]]):
+    if os.path.exists(markdown_path):
+        try:
+            with open(markdown_path, "r") as f:
+                content = f.read()
+
+            if "## References" not in content:
+                content += "\n\n## References\n"
+                for i, citation in citations.items():
+                    content += f"{i}. [{citation['title']}]({citation['url']})\n"
+
+                with open(markdown_path, "w") as f:
+                    f.write(content)
+            else:
+                logger.info(f"References section already exists in {markdown_path}")
+        except Exception as e:
+            logger.error(f"Error adding citations to markdown: {str(e)}", exc_info=True)
+    else:
+        logger.warning(f"Markdown file not found: {markdown_path}")
+
+
+def create_lm_client(
+    model_type, fallback=False, model_settings=None, fallback_model=None
+):
+    try:
+        if model_type not in model_settings:
+            raise ValueError(f"Settings for {model_type} not found")
+
+        if model_type == "ollama":
+            logger.info(
+                f"Creating Ollama client with model: {model_settings['ollama']['model']}"
+            )
+            return OllamaClient(
+                model=model_settings["ollama"]["model"],
+                url="http://localhost",
+                port=int(os.getenv("OLLAMA_PORT", 11434)),
+                max_tokens=model_settings["ollama"]["max_tokens"],
+                stop=("\n\n---",),
+            )
+        elif model_type == "openai":
+            logger.info(
+                f"Creating OpenAI client with model: {model_settings['openai']['model']}"
+            )
+            return OpenAIModel(
+                model=model_settings["openai"]["model"],
+                api_key=os.getenv("OPENAI_API_KEY"),
```

Comment:
Is it possible to use `st.secrets` here? In this way, we don't need to include API key in `.env` again.